### DISCUSSION TITLE
Cleaner way to get a list-based AST

### DISCUSSION THREAD

**Discussion by ThatXliner** [lark-parser/lark#892] [Ideas]
Currently, in order to make a list for an AST component, you'll need to subclass `ast_utils.Ast` and `ast_utils.AsList` to make `ast_utils` stuff a list. Right now, I have to write
```py
@dataclass
class Things(_Ast, ast_utils.AsList):
    content: List[Thing]
```
<!--
```py
class Things(list, _Ast, ast_utils.AsList):
    pass
``` -->
But I wish that instead of making an entirely new dataclass for a list (multiple) of `thing`s, I wanted a function or an extra argument I can pass to `ast_utils.create_transformer`. Actually, I just wanted a list. So what I have right now is actually *more* of a monstrosity.

```py
class Things(list, _Ast, ast_utils.AsList):
    pass
```

I figured I can do some black magic and make a factory function on my own (which should modify the `Transformer` subclass) but this is just an idea to add it to `ast_utils`.

Is there a cleaner way?

**Comment by erezsh** [comment]
Well, you can do

```python
class XlinerList((list, _Ast, ast_utils.AsList):
    pass

class Things(XlinerList):
    pass
```

Doesn't seem that bad.

**Reply by ThatXliner** [reply]
I thought of that already and I didn't want to write a completely new class (for selfish reasons).

Thinking about this overnight, I realized I could've just done
```py
class XlinerList(list, _Ast, ast_utils.AsList):
    pass


Things = XlinerList
```

Which I tested it this morning and it works!

**Reply by erezsh** [reply]
Yep, that too. I thought you wanted to add methods to the object.

I would change the name, though :)

**Reply by ThatXliner** [reply]
>  I thought you wanted to add methods to the object.

What do you mean?
